### ISSUE TITLE
Notice: Trying to access array offset on value of type bool

### ISSUE THREAD

**Issue by danidomen** [PrestaShop/PrestaShop#27774]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Any module or function that uses the native core PrestaShop function "isDiscounted" from the Product class will rise a Notice because an uncontrolled condition inside that function that rely that you have a cart already, but in many cases you will not.

![image](https://user-images.githubusercontent.com/5998908/155170380-183265df-987b-445f-843d-26a64e7aea09.png)

So, the simple thing is to make the condition with if "$ids" is not false (that is the return of evaluating the context cart id_address into the Address::getCountryAndState without really a cart)
![image](https://user-images.githubusercontent.com/5998908/155170816-fbddbf92-16c0-4bf0-a433-f3ec76625330.png)

Of course the "problem" appear when debug mode is on.

Anyway, this produce a lot of lines into the error_log.

### Expected behavior

Not any warning.

### Steps to reproduce

1.- Install a module or create an script that uses the "Product::isDiscounted" function and hooked/executed on a display front hook
1a.- For example https://addons.prestashop.com/es/informes-estadisticas/18739-pixel-plus-eventos-api-de-conversiones-catalogo.html or 
[wim_test27774.zip](https://github.com/PrestaShop/PrestaShop/files/8141510/wim_test27774.zip)

2.- Enable debug mode or tail your error_log

3.- Visit any product of your store without a cart (new visitor).

4.- Execution stopped with debug mode on or a lot of lines into the error log.

### PrestaShop version(s) where the bug happened
1.7.8.0, 1.7.8.2, 1.7.8.3

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

facebookconversiontrackingplus v2.3.3 or any module or script using  Product::isDiscounted

### PR Request ###
https://github.com/PrestaShop/PrestaShop/pull/27775

**Comment by hibatallahAouadni** [comment]
Hello @danidomen 

Could you please provide a basic module contain the "Product::isDiscounted" function to help me to reproduce the issue :pray: 
Waiting for your feedback.

Thanks!

**Comment by danidomen** [comment]
> Hello @danidomen
> 
> Could you please provide a basic module contain the "Product::isDiscounted" function to help me to reproduce the issue 🙏 Waiting for your feedback.
> 
> Thanks!

Sure @hibatallahAouadni ,

Here is it:
[wim_test27774.zip](https://github.com/PrestaShop/PrestaShop/files/8141235/wim_test27774.zip)

Or if you prefer, the code to reproduce it could be something similar to this (on this example hooked on "header"):
```
public function hookHeader($params)
    {
		if($this->context->controller->php_self == 'product' && $idp = (int)Tools::getValue('id_product'))
		{
			$discounted = Product::isDiscounted($idp);
			if($discounted){
				return 'With discount';
			}
			return 'No discount';
		}
    }

```
Remember enable debug mode on your PS and also visit your store product page with a clean session (no cart or logging)

**Comment by khouloudbelguith** [comment]
Hi @danidomen,

thanks for your feedback.
I manage to reproduce the issue with:
1. PS1783 + PHP7.4
2. PS1780+ PHP7.4 
![image](https://user-images.githubusercontent.com/16067358/155722707-20ad5476-dac5-4d2c-956f-867ce662dea2.png)

It is ok with:

1. PS1778 + PHP7.3
2. PS1782 + PHP7.3 
![image](https://user-images.githubusercontent.com/16067358/155722829-3864f636-bf44-4a5b-93b0-cacb11abe449.png)

PS1778 is not compatible with PHP7.4, it is a regression.
so I can confirm that this behavior is indeed a bug.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you!

**Comment by danidomen** [comment]
Hi @khouloudbelguith,

I make the PR also (is on the issue explanation linked also :D https://github.com/PrestaShop/PrestaShop/pull/27775), and indeed I fixed on the projects that we use modules that uses that function. But thanks for the tip ;)

**Comment by Progi1984** [comment]
Fixed by @danidomen in #27922 and tested by @marwachelly
